app/controllers/application_controller/tree_support.rb

```diff
@@ -22,11 +22,15 @@ def squash_toggle
   end
 
   def find_record
-    if %w(container_image host).include? controller_name
-      identify_record(params[:id], controller_name.classify)
-    else
-      identify_record(params[:id], VmOrTemplate)
-    end
+    # TODO: This logic should probably be reversed - fixed list for VmOrTemplate.
+    # (Better yet, override the method only in VmOrTemplate related controllers.)
+    klass = case controller_name
+            when 'host', 'container_replicator', 'container_group', 'container_node', 'container_image'
+              controller_name.classify
+            else
+              VmOrTemplate
+            end
+    identify_record(params[:id], klass)
```

Comment:
I prefer the existing syntax better (mostly because it is a plain string comparison as appose to === done by case)